modules/signal/test/test_signal.cpp

```diff
@@ -0,0 +1,180 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "test_precomp.hpp"
+#include "opencv2/core/mat.hpp"
+#include "opencv2/signal.hpp"
+
+#include <vector>
+#include <numeric>
+#include <cmath>
+#include <string>
+#include <random>
+#include <ctime>
+#include <algorithm>
+
+
+namespace opencv_test { namespace {
+
+using namespace cv;
+using namespace cv::signal;
+
+float MSE(const Mat1f &outSignal, const Mat1f &refSignal)
+{
+    auto mse{0.f};
```

Comment:
Please use

```
float mse = 0.f;
```

here and below (instead of "auto" scalars).

To workaround old ARMv7 compiler.